src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1501,15 +1501,28 @@ async def hook_setup_ha(middleware, *args, **kwargs):
     await middleware.call('failover.status_refresh')
 
     try:
-        ha_configured = await middleware.call(
-            'failover.call_remote', 'failover.status'
-        ) != 'SINGLE'
+        ha_configured = await middleware.call('failover.call_remote', 'failover.status') != 'SINGLE'
+        failover_disabled = 'NO_FAILOVER' in await middleware.call('failover.disabled_reasons')
     except Exception:
-        ha_configured = False
-
-    if ha_configured:
-        # If HA is already configured just sync network
-        if await middleware.call('failover.status') == 'MASTER':
+        ha_configured = failover_disabled = False
+
+    if ha_configured and failover_disabled:
+        # If HA is already configured and failover has been disabled,
+        # and we have gotten to this point, then this means a few things could be happening.
+        #    1. a new interface is being added
+        #    2. an alias is being added to an already configured interface
+        #    3. an interface is being modified (changing vhid/ip etc)
+        #    4. an interface is being deleted
+
+        # In the event #2 happens listed above, there is a race condition that
+        # must be accounted for. When an alias is added to an already configured interface,
+        # a CARP event will be triggered and the interface will go from MASTER to INIT->BACKUP->MASTER which
+        # generates a devd event that is processed by the failover.event plugin.
+        # It takes a few seconds for the kernel to transition the CARP interface from BACKUP->MASTER.
+        # However, we refresh the failover.status while this interface is transitioning.
+        # This means that failover.status returns 'ERROR'. Since we have already checked that there
+        # is a zpool, we simply account for the 'ERROR' status and continue.
+        if 'MASTER' or 'ERROR' in await middleware.call('failover.status'):
```

Comment:
That will be always True because bool('MASTER') is True

Reply:
Also should we run this only if the master flag is true for the given node, to ensure there will be no recursion of calls?